### ISSUE TITLE
Caching example

### ISSUE THREAD

**Issue by danielsjf** [ehossack/django-backblaze-b2#16]
I am looking into using this library but didn't quite get how to configure the caching. I tried to circumvent this setup by ignoring your warning and using the default caching which gave me the error `Missing account data: allowed`.

Would it be possible to include an example in the documentation to understand better how I could configure this?

**Comment by ehossack** [comment]
For sure, let's start with here on this issue. 
Here's a recommended cache setup:

```
CACHES: Dict[str, Any] = {
    "default": {"BACKEND": "django.core.cache.backends.dummy.DummyCache"},
    "django-backblaze-b2": {
        'BACKEND': 'django.core.cache.backends.db.DatabaseCache',
        'LOCATION': 'django_backblaze_b2_cache_table',
    }
}

BACKBLAZE_CONFIG: Dict[str, Any] = {
    "application_key_id": os.environ["BACKBLAZE_APPLICATION_KEY_ID"],
    "application_key": os.environ["BACKBLAZE_APPLICATION_KEY"],
    "bucket": os.environ["BACKBLAZE_BUCKET"],
}
```


This would be a keep-it-minimal cache implementation which does not share state between threads. (Taken from `https://github.com/ehossack/django-backblaze-b2/blob/main/sample_app/sample_app/settings.py#L117`)

```
CACHES: Dict[str, Any] = {
    "default": {"BACKEND": "django.core.cache.backends.dummy.DummyCache"},
    "django-backblaze-b2": {"BACKEND": "django.core.cache.backends.locmem.LocMemCache"},
}

BACKBLAZE_CONFIG: Dict[str, Any] = {
    "application_key_id": os.environ["BACKBLAZE_APPLICATION_KEY_ID"],
    "application_key": os.environ["BACKBLAZE_APPLICATION_KEY"],
    "bucket": os.environ["BACKBLAZE_BUCKET"],
}
```

This would be how to ignore the django caches settings config, and also not share state between threads:
```
# untouched for this example
CACHES: Dict[str, Any] = {
    "default": {"BACKEND": "django.core.cache.backends.dummy.DummyCache"}
}

BACKBLAZE_CONFIG: Dict[str, Any] = {
    "application_key_id": os.environ["BACKBLAZE_APPLICATION_KEY_ID"],
    "application_key": os.environ["BACKBLAZE_APPLICATION_KEY"],
    "bucket": os.environ["BACKBLAZE_BUCKET"],
    "accountInfo": { "type": "memory" }
}
```

Let me know if this makes sense and I can (and should) add a section to the documentation.

**Comment by danielsjf** [comment]
Perfect, never expected an answer so quickly! I completely missed that there was a sample app. That by itself could already be a useful addition to the documentation.

Thanks a lot for those examples. I ultimately went with the third option. According to the [Django documentation](https://docs.djangoproject.com/en/3.2/topics/cache/), memory caching seems like the default recommendation (fastest). It is also less setup than the database caching (albeit just a little ;-) ). Compared to the second I liked that the Backblaze settings are all in one place which makes it easier to keep things separated.

Is there a reason why you went with the default of
```"accountInfo": {"type": "django-cache", "cache": "django-backblaze-b2"}```
instead of
```"accountInfo": { "type": "memory" }```

It seems to me that the memory caching is a good default setup and would allow for less setup on the first implementation.

**Comment by ehossack** [comment]
Heh, I just happened to be on github when you sent this.

To reference the django docs, _Memcached_ is the recommended documentation. It's a bit hard to grasp all the angles (I had to dig deep into this buuuut):

Suppose you deploy django as recommended https://docs.djangoproject.com/en/3.2/howto/deployment/ say using a WSGI method, your application will have `w` workers and `t` threads.

With this configuration:
```
 "accountInfo": { "type": "memory" }
```
every new thread that calls the storage layer will instantiate a new instance of the storage's account info implementation. This means that if you serve 5 requests to the same page simultaneously from a fresh start of your app, you're going to have 5 auth calls and several lookup calls for file info.

Theoretically, multiple threads over the same worker can share access to accountInfo if the implementation is stored in memory managed by django.

So with 
```
 "accountInfo": {"type": "django-cache", "cache": "django-backblaze-b2"}
```

The library can use django's in memory cache implementation, and save the file lookup info at least, even if across workers there's no savings, that's still savings on the threads per workers.

When you use a cache that exists outside of memory (memcached/redis/database) then multiple workers can share access to this resource, and caching happens at the auth layer + file lookup layer across threads.

So yeah, a good default setup would be nice, but it's oddly nuanced, and I hestitate recommending a method that comes with such performance limitations. 

In particular, when I used the config of:
```
 "accountInfo": { "type": "memory" }
```
myself in production, I ran into issues where threads might access an uninitialized accountInfo object from somewhere in the process, causing errors, and the python b2 team said that the implementation was not threadsafe 😂

**Comment by danielsjf** [comment]
Alright that's a good addition and I will change how I use it as well.

In that case I think for future users in my situation, these few lines of the cache definition would help a lot to get started.

```
CACHES: Dict[str, Any] = {
    "default": {"BACKEND": "django.core.cache.backends.dummy.DummyCache"},
    "django-backblaze-b2": {"BACKEND": "django.core.cache.backends.locmem.LocMemCache"},
}
```

If needed you can even add a link to this issue for the interested reader.

**Comment by ehossack** [comment]
Addressed in f04a5b1
